WordPress/Classes/ViewRelated/Blog/SiteManagement/SettingsTitleSubtitleController.swift

```diff
@@ -0,0 +1,283 @@
+            case .description:
+                return NSLocalizedString("Description", comment: "Section header for tag name in Tag Details View.").uppercased()
+            }
+        }
+
+        var height: CGFloat {
+            switch self {
+            case .name:
+                return WPTableViewDefaultRowHeight
+            case .description:
+                return WPTableViewDefaultRowHeight * 3
+            }
+        }
+    }
+
+    private lazy var nameCell: WPTableViewCell = {
+        return self.cell(content: self.nameTextField)
+    }()
+
+    private lazy var descriptionCell: WPTableViewCell = {
+        return self.cell(content: self.descriptionTextField)
+    }()
+
+    private lazy var nameTextField: UITextField = {
+        return self.textField()
+    }()
+
+    private lazy var descriptionTextField: UITextView = {
+        return self.textView()
+    }()
+
+    private let data: SettingsTitleSubtitleController.Data
+    private let confirmation: SettingsTitleSubtitleController.Confirmation?
+
+    private var action: SettingsTitleSubtitleAction?
+    private var update: SettingsTitleSubtitleAction?
+    private var isTriggeringAction = false
+
+    public init(data: SettingsTitleSubtitleController.Data, confirmation: SettingsTitleSubtitleController.Confirmation? = nil) {
+        self.data = data
+        self.confirmation = confirmation
+        super.init(style: .grouped)
+    }
+
+
+    /// Closure to be executed when the right bar button item is tapped. If there was a Confirmation passed in this VC's constructor, this closure will be called only after users confirm an alert.
+    ///
+    func setAction(_ closure: @escaping SettingsTitleSubtitleAction) {
+        action = closure
+    }
+
+
+    /// Closure to be executed when the users tap the Back button, only if there is valid text in the Title textfield
+    ///
+    func setUpdate(_ closure: @escaping SettingsTitleSubtitleAction) {
+        update = closure
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        setupNavigationBar()
+        setupTitle()
+        setupTable()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        validateData()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        isTriggeringAction = false
+    }
+
+    private func setupNavigationBar() {
+        guard let title = data.title, title.count > 0 else {
+            return
+        }
+
+        navigationItem.rightBarButtonItem = deleteButton()
+    }
+
+    private func deleteButton() -> UIBarButtonItem {
+        let trashIcon = Gridicon.iconOfType(.trash)
+        return UIBarButtonItem(image: trashIcon, style: .plain, target: self, action: #selector(deleteContent))
```

Comment:
The controller displays a `.trash` icon, and has methods named `deleteButton` and `deleteContent`.   This implies the action will always be a delete action.  Would make sense to either make the icon configurable, and rename the methods, or to make it clear the expected action is always a delete?